packages/react-components/react-card/src/components/Card/Card.cy.tsx

```diff
@@ -240,4 +260,115 @@ describe('Card', () => {
+
+        return <CardSample onCardSelect={onCardSelect} />;
+      };
+
+      mountFluent(<Example />);
+
+      cy.get('#card').should('have.attr', 'role', 'checkbox');
+    });
+
+    it('should have internal checkbox when selectable - onCardSelect prop', () => {
+      mountFluent(<CardSample select={<span />} />);
+
+      cy.get(`.${cardClassNames.select}`).should('exist');
+      cy.get(`.${cardClassNames.root}`).should('not.have.attr', 'role', 'checkbox');
+    });
+
+    it('should render checkbox when passing a custom slot', () => {
+      mountFluent(<CardSample select={<input type="checkbox" />} />);
+
+      cy.get(`.${cardClassNames.select} input[type="checkbox"]`).should('exist');
+    });
+
+    it('should select with a mouse click', () => {
+      mountFluent(<CardSample defaultSelected={false} />);
+
+      cy.get(`.${cardClassNames.root}`).should('have.attr', 'aria-checked', 'false');
+      cy.get(`.${cardClassNames.root}`).realClick();
+      cy.get(`.${cardClassNames.root}`).should('have.attr', 'aria-checked', 'true');
+    });
+
+    it('should have checkbox pre-selected and toggle its value', () => {
+      mountFluent(<CardSample defaultSelected />);
+
+      cy.get(`.${cardClassNames.root}`).should('have.attr', 'aria-checked', 'true');
+      cy.get(`.${cardClassNames.root}`).realClick();
+      cy.get(`.${cardClassNames.root}`).should('have.attr', 'aria-checked', 'false');
+    });
+
+    it('should select with the Space key', () => {
+      mountFluent(<CardSample defaultSelected={false} />);
+
+      cy.get(`.${cardClassNames.root}`).focus().realPress('Space');
+      cy.get(`.${cardClassNames.root}`).should('have.attr', 'aria-checked', 'true');
+    });
+
+    it('should NOT select with the Enter key if card has any actions inside', () => {
+      mountFluent(<CardSample defaultSelected={false} />);
+
+      cy.get(`.${cardClassNames.root}`).focus().realPress('Enter');
+      cy.get(`.${cardClassNames.root}`).should('have.attr', 'aria-checked', 'false');
+      cy.get(`.${cardClassNames.root} button`).first().should('be.focused');
+    });
+
+    it('should NOT select when focused on an action', () => {
+      mountFluent(<CardSample defaultSelected={false} />);
+
+      cy.get(`.${cardClassNames.root} button`).first().focus().realPress('Enter');
+      cy.get(`.${cardClassNames.root}`).should('have.attr', 'aria-checked', 'false');
+    });
+
+    it('should select with the Enter key if card has any actions inside', () => {
+      mountFluent(<CardSampleNoActions defaultSelected={false} />);
+
+      cy.get(`.${cardClassNames.root}`).focus().realPress('Enter');
+      cy.get(`.${cardClassNames.root}`).should('have.attr', 'aria-checked', 'true');
+    });
+
+    it('should sync selected value with custom slot', () => {
+      const Example = () => {
+        const [checked, setChecked] = React.useState(false);
+
+        const onCardSelect = React.useCallback((event, { selected }) => setChecked(selected), []);
```

Comment:
I don't think you need to use `useCallack` for this.

```suggestion
        const onCardSelect = (event, { selected }) => setChecked(selected);
```